### PR TITLE
chore: config changes & cleanups

### PR DESCRIPTION
**Pull request by fxck**
https://github.com/katallaxie/angular2-preboot/issues/56 did 2., 3., 4. and some cleanups

what do you think @katallaxie

### CODE REVIEW

File: config/webpack.d.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+export declare const EXCLUDE_SOURCEMAPS: any[];
+export declare const COPY_FOLDERS: any[];
+export declare const PLUGINS: {
+  common: any[];
```
Comment (katallaxie) [inline config/webpack.d.ts]: should we make them uppercase?!

File: src/index.html [hunk 2]
```diff
@@ -7,8 +7,6 @@
 
   <title><%= htmlWebpackPlugin.options.meta.title %></title>
 
-  <meta name="description" content="<%= htmlWebpackPlugin.options.meta.title %>">
```
Comment (katallaxie) [inline src/index.html]: 👍

File: config/webpack.d.ts (hunk 1 above)
Comment (fxck) [inline config/webpack.d.ts]: personally, I wouldn't `PLUGINS` is the constant, common, dev, prod are just are properties

File: webpack.config.ts [hunk 3]
```diff
@@ -93,31 +87,83 @@ const COPY_FOLDERS = [
...
 
 ];
 
 // is dll
 if (!isDll && isDev) {
-  tryDll(['polyfills', 'vendors', 'rxjs']);
+  tryDll([ 'polyfills', 'vendors', 'rxjs' ]);
 }
 
+let loaderRules = {
```
Comment (katallaxie) [inline webpack.config.ts]: Can we keep these seperated?!

File: config/webpack.d.ts (hunk 1 above)
Comment (katallaxie) [inline config/webpack.d.ts]: should we make them uppercase?!

File: webpack.config.ts (hunk 3 above)
Comment (fxck) [inline webpack.config.ts]: Well, I guess, but what's the reason you separated them? It just feels like a thing you would not really change ever(on per project basis anyway) and some parts of `webpack.config.ts` excepts some of the rules to exist, or be written in a particular way, so the user can't just go any randomly delete or change them..

Comment (fxck) [inline webpack.config.ts]: What I thought about was creating another property of `PLUGINS`, something like `default`, which would have interface looking something like this 

```
PLUGINS: {
  default: {
    cssLoader?: {
      test?: any;
      use?: any;
      exclude?: any;
      options?: any;
    }
  }
}
```

which would then override default settings of the individual default rules without polluting the config folder with a 100LOC file... if you really wanted to do that.. but then again, it really seems like an overkill for the reason I've written above.

@katallaxie

File: config/webpack.d.ts (hunk 1 above)
Comment (fxck) [inline config/webpack.d.ts]: personally, I wouldn't, PLUGINS is the constant, common, dev, prod are just are properties

File: webpack.config.ts (hunk 3 above)
Comment (fxck) [inline webpack.config.ts]: and by passing in `false`, you could disable the rule(at least on those where it wouldn't break something)

**Comment by katallaxie** [comment]
I have been thus far a bit 🍒 picking with your ideas.

**Comment by fxck** [comment]
closing in favor of https://github.com/katallaxie/angular2-preboot/pull/59

**Pull request closed without merging**